src/EventStore.ClientAPI/Internal/ClusterDnsEndPointDiscoverer.cs

```diff
@@ -177,11 +177,24 @@ internal class ClusterDnsEndPointDiscoverer : IEndPointDiscoverer {
 				url,
 				null,
 				response => {
-					if (response.HttpStatusCode != HttpStatusCode.OK) {
-						_log.Info("[{0}] responded with {1} ({2})", endPoint, response.HttpStatusCode,
-							response.StatusDescription);
+					try {
+						if (response.HttpStatusCode != HttpStatusCode.OK) {
+							_log.Info("[{0}] responded with {1} ({2})", endPoint.EndPoint, response.HttpStatusCode,
+								response.StatusDescription);
+							return;
+						}
+
+						try {
+							result = response.Body.ParseJson<ClusterMessages.ClusterInfoDto>();
+							//_log.Debug("ClusterDnsEndPointDiscoverer: Got gossip from [{0}]:\n{1}.", endPoint, string.Join("\n", result.Members.Select(x => x.ToString())));
+						} catch (Exception e) {
+							if (e is AggregateException ae)
+								e = ae.Flatten();
+							_log.Error("Failed to get cluster info from [{0}]: deserialization error: {1}.",
+								endPoint.EndPoint, e);
+						}
+					} finally {
 						completed.Set();
-						return;
 					}
 
 					try {
```

Comment:
This try-catch-finally block seems redundant (same as above). It is not present on master.